### PR TITLE
Renovation: implement Button.stylingMode

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

File: js/renovation/button.tsx [hunk 1]
```diff
@@ -1,56 +1,35 @@
...
-
-    if (model.stylingMode === 'outlined') {
-        classNames.push('dx-button-mode-outlined');
-    } else if (model.stylingMode === 'text') {
-        classNames.push('dx-button-mode-text');
-    } else {
-        classNames.push('dx-button-mode-contained');
-    }
+    const classNames = ['dx-button'].concat(model.classNames);
+    const isValidStylingMode = ['outlined', 'text', 'contained'].indexOf(model.stylingMode) !== -1;
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx]: We can extract these string arrays instead of creating them each time

**Review by roman-simionov — DISMISSED** [review]

**Review by LazyLahtak — APPROVED** [review]

File: js/renovation/button.tsx [hunk 2]
```diff
@@ -16,16 +16,18 @@ const getImageContainerJSX = (source: string) => {
 };
 
 const stylingModes = ['outlined', 'text', 'contained'];
+const defaultClassNames = ['dx-button'];
 
 const getCssClasses = (model: any) => {
-    const classNames = ['dx-button'].concat(model.classNames);
-    const isValidStylingMode = stylingModes.indexOf(model.stylingMode) !== -1;
+    const { text, icon, stylingMode, type } = model;
```
Comment (MaximKudriavtsev) [inline js/renovation/button.tsx:22]: We can use argument destruction here

```js
const getCssClasses = ({ text, icon, stylingMode, type })
```

Comment (churkin) [reply js/renovation/button.tsx:22]: We will miss 'classNames'

Comment (MaximKudriavtsev) [reply js/renovation/button.tsx:22]: We can use alias name
```js
const getCssClasses = ({ text, icon, stylingMode, type, classNames: classes })
```

**Review by MaximKudriavtsev — APPROVED** [review]

**Pull request merged**